docroot/themes/custom/uids_base/uids_base.theme

```diff
@@ -314,6 +312,63 @@ function uids_base_preprocess_block(&$variables) {
 
 }
 
+/**
+ * Implements hook_preprocess_HOOK() for block_content__uiowa_collection__tab.html.twig.
+ */
+function uids_base_preprocess_block_content__uiowa_collection__tab(&$variables) {
+  $variables['tabs'] = [];
+  $items = $variables['elements']['field_uiowa_collection_items']['#items'];
+  foreach ($items as $delta => $item) {
+    $id = $item->get('target_id')->getString();
+    $paragraph = \Drupal::service('entity_type.manager')
+      ->getStorage('paragraph')
+      ->load($id);
+    $label = $paragraph->get('field_collection_headline')->getString();
+    if ($paragraph->get('field_collection_item_view_mode') !== 'tab') {
+      $paragraph->set('field_collection_item_view_mode', 'tab')
+        ->save();
+    }
+    $variables['tabs'][$delta] = [
+      'id' => $id,
+      'label' => $label,
+    ];
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK() for block_content__uiowa_collection__accordion.html.twig.
+ */
+function uids_base_preprocess_block_content__uiowa_collection__accordion(&$variables) {
+  $items = $variables['elements']['field_uiowa_collection_items']['#items'];
+  foreach ($items as $delta => $item) {
+    $id = $item->get('target_id')->getString();
+    $paragraph = \Drupal::service('entity_type.manager')
+      ->getStorage('paragraph')
+      ->load($id);
+    if ($paragraph->get('field_collection_item_view_mode') !== 'accordion') {
+      $paragraph->set('field_collection_item_view_mode', 'accordion')
+        ->save();
```

Comment:
It looks like you are saving the paragraph in theme preprocess function, which seems like it would run quite a bit more than you would want it to.

Reply:
Initial thought was to have the check before so it would only run/save once when the block view_mode is set (or updated). As long as the page is being edited through the UI, this save should run as part of the rendering process of the layout editor, keeping the main saving cost upfront on the editor.

I think you could get away with not saving at all, but setting the field each time--it would save us a save (ha) that's more costly, but need to run and set the variable in every preprocess. Not sure which is more efficient in the long run, especially getting into pages like https://medcom.uiowa.edu/theloop/covid-19-clinical-information (not ours, not Drupal, but good example of how many paragraphs we might end up with on a page).

Ideally, I'd rather this live with the block creation/update form submission, but haven't been able to nail that down--the block creation happens in a process that gets called after the submission itself that I haven't found a good way to hook into.

Reply:
Child view modes and a bit of refactoring of how the templates were laid out worked like a charm! 👍  Cleaned things up, and was able to then remove all these theme preprocess hooks but for the one handling tab labels.